### PR TITLE
added 'not' to filter types

### PR DESCRIPTION
**Pull request by hsalhab**
Added "not" to FILTER_TYPES, allowing the use of a not filter to negate other boolean filters

### CODE REVIEW

File: src/druidry/data_source.py [hunk 1]
```diff
@@ -322,6 +322,13 @@ def combine_filter(f):
         return OrFilter(fields=filters)
 
 
+def negate_filter(f):
+    filter = translate_filter(f['filter'])
```
Comment (joesolly) [inline src/druidry/data_source.py]: please use a different variable name like `f` as `filter` is a reserved keyword in python

**Review by joesolly — CHANGES_REQUESTED** [review]
can you add a test to this class too https://github.com/monetate/druidry/blob/63855ee7c05dc429054788b5060cc59767bf7bf2/tests/test_data_source.py#L169

File: src/druidry/data_source.py (hunk 1 above)
Comment (hsalhab) [reply src/druidry/data_source.py]: resolved

File: src/druidry/data_source.py [hunk 2]
```diff
@@ -322,6 +322,13 @@ def combine_filter(f):
         return OrFilter(fields=filters)
 
 
+def negate_filter(f):
+    filter_ = translate_filter(f['filter'])
```
Comment (joesolly) [inline src/druidry/data_source.py]: https://realpython.com/python-pep8/#naming-styles
let's try for something like `translated_filter`

Comment (hsalhab) [reply src/druidry/data_source.py]: fixed :) I was trying to follow naming convention in other functions in the same file as well as [https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles)

**Review by joesolly — APPROVED** [review]

**Pull request merged**